lib/metasploit/framework/data_service/remote/http/remote_async_callback_service.rb

```diff
@@ -0,0 +1,17 @@
+require 'metasploit/framework/data_service/remote/http/response_data_helper'
+
+module RemoteAsyncCallbackDataService
+  include ResponseDataHelper
+
+  ASYNC_CALLBACK_API_PATH = '/api/v1/async-callbacks'
+  ASYNC_CALLBACK_MDM_CLASS = 'Mdm::AsyncCallback'
+
+  def async_callbacks(opts)
+    path = get_path_select(opts, ASYNC_CALLBACK_API_PATH)
+    json_to_mdm_object(self.get_data(path, nil), ASYNC_CALLBACK_MDM_CLASS, [])
+  end
+
+  def create_async_callback(opts)
+    json_to_mdm_object(self.post_data(ASYNC_CALLBACK_API_PATH), ASYNC_CALLBACK_MDM_CLASS, []).first
```

Comment:
The 2nd parameter to `post_data` was incorrectly removed when it should have been the 3rd parameter to `json_to_mdm_object`, the `[]`.